subt/cloudsim2osgar.py

```diff
@@ -133,7 +137,11 @@ def __init__(self, robot_name):
             channel, data = self.bus.listen()
             print("_"*50)
             print("receiving:", data)
-            # TODO: switch on channel to feed different ROS publishers
+            # switch on channel to feed different ROS publishers
+            if channel in publishers:
+                publishers[channel].publish(*data)  # just guessing for Empty, where ([],) is wrong
```

Comment:
What is ` # just guessing for Empty, where ([],) is wrong`?

Reply:
I can remove the comment - I was not able to correctly publish the data (the Empty data) ... and the listen() function returned `([],)` instead of `[]`, so that's why I added *

Reply:
Are you sure that `listen()` really returns `([],)` when `[]` is published? That would be a bug in need of fixing.

<pre>&gt;&gt;&gt; import msgpack
&gt;&gt;&gt; msgpack.packb([], use_bin_type=True)
&apos;\x90&apos;
&gt;&gt;&gt; msgpack.unpackb(&apos;\x90&apos;, raw=False)
[]
&gt;&gt;&gt; 
</pre>

It is not that simple to verify in a real run since one side is Python3 but msgpack behaves in both pythons the same.

Oh, now I get it:

<pre>&gt;&gt;&gt; Empty([])
Traceback (most recent call last):
  File &quot;&lt;stdin&gt;&quot;, line 1, in &lt;module&gt;
  File &quot;/opt/ros/melodic/lib/python2.7/dist-packages/std_msgs/msg/_Empty.py&quot;, line 33, in __init__
    super(Empty, self).__init__(*args, **kwds)
  File &quot;/opt/ros/melodic/lib/python2.7/dist-packages/genpy/message.py&quot;, line 344, in __init__
    raise TypeError(&apos;Invalid number of arguments, args should be %s&apos; % str(self.__slots__) + &apos; args are&apos; + str(args))
TypeError: Invalid number of arguments, args should be [] args are([],)
</pre>

So what we are really doing here is encoding the arguments to `__init__` of different ROS msgs as a list. I am totally OK with that.

Let's delete the comment in some future pass since it is confusing.

Reply:
OK, thanks for investigation :+1: